lib/parseBinaryGltf.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+var fs = require('fs');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var StringDecoder = require('string_decoder').StringDecoder;
+module.exports = parseBinaryGltf;
+
+function parseBinaryGltf(data) {
+    var sizeOfUint32 = Uint32Array.BYTES_PER_ELEMENT;
+    var byteOffset = 0;
+
+    if (data.slice(byteOffset, sizeOfUint32).toString() !== 'glTF') {
+        throw new Error('File is not valid binary glTF');
+    }
+
+    byteOffset += 3*sizeOfUint32;
+    var dataView = new DataView(data.buffer);
+    var sceneLength = dataView.getUint32(byteOffset, true);
+
+    byteOffset += 2*sizeOfUint32;
+    var decoder = new StringDecoder();
+    var scene = decoder.write(data.slice(byteOffset, byteOffset + sceneLength));
+    var gltf = JSON.parse(scene);
+
+    byteOffset += sceneLength;
+    var body = data.slice(byteOffset);

```

Comment:
With the above said, this `slice` is probably what we want here and the `slice` above is for a small enough buffer that we might not care.